beater/connections.go

```diff
@@ -89,19 +93,21 @@ func process(ps *processes.Processes, exposeProcessInfo bool, inode uint64) *pro
 }
 
 func filterAndPublish(exposeProcessInfo, exposeCmdline, exposeEnviron bool, aggregation time.Duration, socketInfo <-chan *sockets.SocketInfo, connections chan<- Connection, servers chan ServerConnection) {
-	listeningOn := make(map[uint16]time.Time)
+	listeningOn := make(map[incomingConnectionDedup]time.Time)
 	outgoingConnectionSeen := make(map[outgoingConnectionDedup]time.Time)
 	ps := processes.New(exposeCmdline, exposeEnviron)
 
 	for {
 		now := time.Now()
 		select {
 		case s := <-socketInfo:
-			if when, seen := listeningOn[s.SrcPort]; !seen || now.Sub(when) > aggregation {
+			localIP := s.SrcIP.String()
```

Comment:
Ok, if I understand it correctly previously you filtered based on the sourcePort, which could have filtered out an ip address, if there are two ip addresses with the same port. And now you dedup on both ip and port, right? Code looks good as far as I can understand it.